batched/dense/src/KokkosBatched_Gemm_Decl.hpp

```diff
@@ -259,6 +259,42 @@ template <class ArgTransA, class ArgTransB, class ArgBatchSzDim,
           int tile_m, int tile_n, int tile_k>
 class BatchedDblBufGemm;
 
+//////////////////////////////// tile_m //////////////////////////////////
+template <typename ExecutionSpace>
+constexpr KOKKOS_INLINE_FUNCTION int kk_gemm_dlb_buf_tile_m() {
+  return 32;
+}
+//////////////////////////////// tile_n //////////////////////////////////
+template <typename ExecutionSpace>
+constexpr KOKKOS_INLINE_FUNCTION int kk_gemm_dlb_buf_tile_n() {
+  return 32;
+}
+//////////////////////////////// tile_k //////////////////////////////////
+template <typename ExecutionSpace>
+constexpr KOKKOS_INLINE_FUNCTION int kk_gemm_dlb_buf_tile_k() {
+  return 8;
+}
+
+// On MI100, batched_scalar_batched_gemm_nt_nt_dcomplex_dcomplex_right fails
+// without this. See https://github.com/kokkos/kokkos-kernels/issues/1547.
+// This reduces the register allocations (REG_M and REG_N) in the double
+// buffering algorithm by a factor of 2.
+#if defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_ARCH_VEGA908)
+template <>
+constexpr KOKKOS_INLINE_FUNCTION int
+kk_gemm_dlb_buf_tile_k<Kokkos::Experimental::HIP>() {
+  return 16;
```

Comment:
@e10harvey I don't really follow. Please remind me. Why can tile_k=16 for HIP reduce the register allocation as compared to tile_k=8? Isn't that with tile_k=16 we have larger tile which needs more register? Am I missing something?

Reply:
`tile_k = 16` results in larger shared memory allocations but smaller REG_M and REG_N sizes:
https://github.com/kokkos/kokkos-kernels/blob/f3556c5c9791c3bc6e8f9d6e8c6b03d7322cc15a/batched/dense/impl/KokkosBatched_Gemm_DblBuf_Impl.hpp#L385-L386

https://github.com/kokkos/kokkos-kernels/blob/f3556c5c9791c3bc6e8f9d6e8c6b03d7322cc15a/batched/dense/impl/KokkosBatched_Gemm_DblBuf_Impl.hpp#L146-L147

Reply:
Thanks @e10harvey. Now I remember. :-)